compiler/rustc_privacy/src/lib.rs

```diff
@@ -149,7 +149,7 @@ where
             {
                 let tcx = self.def_id_visitor.tcx();
                 if let Ok(Some(ct)) = AbstractConst::new(tcx, uv) {
-                    self.visit_abstract_const_expr(tcx, ct)?;
+                    ct.visit_with(self)?;
```

Comment:
i think we should change `PredicateKind:ConstEvaluatable` to use a `ty::Const` instead of `Unevaluated`.

Because this current code feels quite unfortunate to me. I guess its fine for now 🤷

Reply:
I can do it if it would make it more legible, I just modified it so that it works w/o thinking too much about further implications

Reply:
would be nice, but that should be done in a separate pr then